projects/packages/google-fonts-provider/src/class-utils.php

```diff
@@ -44,14 +44,40 @@ public static function font_source_resource_hint( $urls, $relation_type ) {
 	 * @return boolean|void Whether the font family is registered, or void if WP_Webfonts is not available.
 	 */
 	public static function is_font_family_registered( $font_family_name ) {
-		if ( ! function_exists( 'wp_webfonts' ) || ! method_exists( 'WP_Webfonts', 'get_font_slug' ) ) {
-			return;
-		}
+		// New WP Fonts API since Gutenberg 14.9
+		// See https://github.com/Automattic/jetpack/issues/28063
+		if ( class_exists( 'WP_Fonts' ) ) {
+			if (
+				! function_exists( 'wp_webfonts' ) ||
+				! class_exists( 'WP_Web_Fonts' ) ||
+				! class_exists( 'WP_Webfonts_Utils' )
+			) {
+				return;
+			}
+
+			$wp_webfonts = wp_webfonts();
+
+			if ( ! method_exists( $wp_webfonts, 'get_registered_font_families' ) ) {
+				return;
+			}
+
+			$handle = \WP_Webfonts_Utils::convert_font_family_into_handle( $font_family_name );
```

Comment:
TODO: rename to `WP_Fonts_Utils` after API rename in https://github.com/WordPress/gutenberg/pull/46749